types/functional-red-black-tree/index.d.ts

```diff
@@ -0,0 +1,190 @@
+// Type definitions for functional-red-black-tree 1.0
+// Project: https://github.com/mikolalysenko/functional-red-black-tree
+// Definitions by: Campbell Wass <https://github.com/CamWass>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace createRBTree {
+    /** Represents a functional red-black tree. */
+    interface Tree<K, V> {
+        /** Returns the root node of the tree. */
+        root: Node<K, V>;
+
+        /** A sorted array of all keys in the tree. */
+        readonly keys: K[];
```

Comment:
Here and elsewhere, I've made certain properties readonly because the library defines them as such:
```javascript
Object.defineProperty(obj.prototype, "key", {
  get: function() {
    // return some value
  },
  enumerable: true
})
```
My understanding is that attempting to set these properties like below will have no effect:
```javascript
const myObj = new obj();
const initial = myObj.key; // for argument's sake, pretend this is a number
myObj.key = "abc";
myObj.key === initial // will be true because the property is unchanged
```
Is this understanding correct, and is it safe to mark properties defined in this way as readonly?